ts/index.ts

```diff
@@ -0,0 +1,196 @@
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.rolling = inputs.getBool("rolling");
+    this.mirror = inputs.getBool("mirror");
+    this.name = inputs.getStringOrNull("name");
+    this.rollingMinor = inputs.getNumberOrNull("rolling-minor");
+
+    // Other inputs
+    this.sourceBinary = this.determineSourceBinary;
+  }
+
+  // We first check for a value using the `source-binary` input; if
+  // now-deprecated `flakehub-push-binary`
+  private get determineSourceBinary(): string | null {
+    const sourceBinaryInput = inputs.getStringOrNull("source-binary");
+    const flakeHubPushBinaryInput = inputs.getStringOrNull(
+      "flakehub-push-binary",
+    );
+
+    return sourceBinaryInput !== ""
+      ? sourceBinaryInput
+      : flakeHubPushBinaryInput;
+  }
+
+  private determineVisibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      actionsCore.setFailed(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.map((opt) => `\`${opt}\``).join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
+    const env: ExecutionEnvironment = {};
+
+    env.FLAKEHUB_PUSH_VISIBILITY = this.visibility;
+    env.FLAKEHUB_PUSH_TAG = this.tag;
+    env.FLAKEHUB_PUSH_HOST = this.host;
+    env.FLAKEHUB_PUSH_LOG_DIRECTIVES = this.logDirectives;
+    env.FLAKEHUB_PUSH_LOGGER = this.logger;
+    env.FLAKEHUB_PUSH_GITHUB_TOKEN = this.gitHubToken;
+    env.FLAKEHUB_PUSH_REPOSITORY = this.repository;
+    env.FLAKEHUB_PUSH_DIRECTORY = this.directory;
+    env.FLAKEHUB_PUSH_GIT_ROOT = this.gitRoot;
+    // not included: the now-deprecated FLAKEHUB_PUSH_EXTRA_TAGS
+    env.FLAKEHUB_PUSH_EXTRA_LABELS = this.extraLabels;
+    env.FLAKEHUB_PUSH_SPDX_EXPRESSION = this.spdxExpression;
+    env.FLAKEHUB_PUSH_ERROR_ON_CONFLICT = this.errorOnConflict.toString();
+    env.FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS = this.includeOutputPaths.toString();
+    env.FLAKEHUB_PUSH_ROLLING = this.rolling.toString();
+    env.FLAKEHUB_PUSH_MIRROR = this.mirror.toString();
+
+    if (this.name !== null) {
+      env.FLAKEHUB_PUSH_NAME = this.name;
+    }
+
+    if (this.rollingMinor !== null) {
+      env.FLAKEHUB_PUSH_ROLLING_MINOR = this.rollingMinor.toString();
+    }
+
+    return env;
+  }
+
+  async push(): Promise<void> {
+    const executionEnv = await this.executionEnvironment();
+
+    const binary =
+      this.sourceBinary !== null
+        ? this.sourceBinary
+        : await this.idslib.fetchExecutable();
```

Comment:
maybe it's somehow inside fetchExecutable, but this doesn't look like it handles:
* https://github.com/DeterminateSystems/flakehub-push/blob/main/action.yaml#L84-L99
* (and the diff shows deleted code that set a URL that would've been fetched previously on the fly)

Reply:
(I am personally sort of in favor of maybe simplifying those to a *single* `flakehub-push-binary-url` option, and expect the user to be able to form that URL from wherever they want it pulled from, but it is nice to be able to test (for example) a heavily refactored flakehub-action binary in some existing workflow.)

Reply:
https://github.com/DeterminateSystems/flakehub-push/pull/124#issuecomment-2071221447

Reply:
This is actually handled by `fetchExecutable`, which I need to write up some docs for. Basically it uses the `source-*` params to construct the URL. It even handles having a backup prefix (in this case `flakehub-push-*`) for deprecated options.

Reply:
oops, also, nice.

Reply:
Notice this setting: https://github.com/DeterminateSystems/flakehub-push/blob/detsys-ts/ts/index.ts#L63

That instructs the `fetchExecutable` function to use `source-*` and then `flakehub-push-*` as a backup in each case.